tempest/api/workloadmgr/multiattach_volumes/test_multiattach_volumes.py

```diff
@@ -524,20 +530,219 @@ def test_01_multiattach_volumes(self):
+                                           "destination_type": "volume"}]
+            self.vm_id = self.create_vm(
+                key_pair=self.kp,
+                image_id="",
+                block_mapping_data=self.block_mapping_details)
+
+            # Now create volume with derived volume type id...
+            self.volume_id = self.create_volume(
+                volume_type_id=vol_type_id, size=6)
+
+            LOG.debug("Volume ID: " + str(self.volume_id))
+
+            self.volumes.append(self.volume_id)
+            # Attach volume to vm...
+            self.attach_volume(self.volume_id, self.vm_id)
+            LOG.debug("Multiattach Volume attached to vm: " + str(self.vm_id))
+            api_version.COMPUTE_MICROVERSION = None
+
+            fip = self.get_floating_ips()
+            LOG.debug("\nAvailable floating ips are {}: \n".format(fip))
+
+            if len(fip) < 3:
+                raise Exception("Floating ips unavailable")
+            self.set_floating_ip(fip[0], self.vm_id)
+
+            md5sums_before_full = self._add_data_on_instance_and_volume([fip[0]])
+            LOG.debug(
+                f"md5sums_before_full: {md5sums_before_full}")
+
+            # Create workload with API
             try:
-                self.detach_volume(self.vm_id_1, self.volume_id)
-            except:
-                pass
+                self.wid = self.workload_create([self.vm_id],
+                                                tvaultconf.workload_type_id)
+                LOG.debug("Workload ID: " + str(self.wid))
+            except Exception as e:
+                LOG.error(f"Exception: {e}")
+                raise Exception("Create workload with volume booted vm")
+            if (self.wid is not None):
+                self.wait_for_workload_tobe_available(self.wid)
+                self.workload_status = self.getWorkloadStatus(self.wid)
+                if (self.workload_status == "available"):
+                    reporting.add_test_step("Create workload with volume booted vm", tvaultconf.PASS)
+                    tests[0][1] = 1
+                    reporting.test_case_to_write()
+                else:
+                    raise Exception("Create workload with volume booted vm")
+            else:
+                raise Exception("Create workload with volume booted vm")
+
+            reporting.add_test_script(tests[1][0])
+            self.snapshot_id = self.workload_snapshot(self.wid, True)
+            self.wait_for_workload_tobe_available(self.wid)
+            self.snapshot_status = self.getSnapshotStatus(self.wid,
+                                                          self.snapshot_id)
+            if (self.snapshot_status == "available"):
+                reporting.add_test_step("Create full snapshot", tvaultconf.PASS)
+                self.mount_path = self.get_mountpoint_path(tvaultconf.tvault_ip[0], tvaultconf.tvault_username,
```

Comment:
Updated code looks good. Just try to take this (get_mountpoint_path) out of if condition. It will help us to keep code formation as same for both the if conditions ( full snapshot & incremental snapshot status check ).